polyfactory/value_generators/constrained_collections.py

```diff
@@ -39,6 +40,14 @@ def handle_constrained_collection(
     min_items = abs(min_items if min_items is not None else (max_items or 0))
     max_items = abs(max_items if max_items is not None else min_items + 1)
 
+    if collection_type in (frozenset, set) or unique_items:
+        if hasattr(field_meta.annotation, "__origin__") and field_meta.annotation.__origin__ is Literal:
+            min_items = 1
+            max_items = 1
```

Comment:
Why is `max_items` set to `1` here? A literal could have more than one value no? Like `Literal["foo", "bar", "baz"]`?

Reply:
Ah, I see where I was confused. `min_items` is also wrong now that 0-length collections are allowed.

I parameterized the tests a bit better and also found bugs when `max_items` is less than the number of possible values, so I ended up rewriting most of the logic here.